content/doc/developer/plugin-development/build-process.adoc

```diff
@@ -0,0 +1,28 @@
+---
+layout: developersection
+title: Plugin Build Process
+references:
+- url: ../updating-parent/
+  title: Guide to Updating Your Maven Parent POM
+- url: https://github.com/jenkinsci/plugin-pom/
+  title: Plugin parent POM 2.x and up
+- url: https://github.com/jenkinsci/maven-hpi-plugin/
+  title: Maven HPI Plugin
+---
+
+Most Jenkins plugins are built with link:https://maven.apache.org[Maven]. The link:https://github.com/jenkinsci/maven-hpi-plugin/[Maven HPI plugin] does the heavy lifting, such as bundling plugins in the HPI/JPI archive format, or running a debug Jenkins instance with the plugin.
+
+Jenkins plugins typically use the Jenkins plugin parent POM, which provides a sensible default configuration for the build.
+Up to Jenkins 1.645, the plugin POM was kept in sync with Jenkins releases, so that the minimum required Jenkins version for a plugin determined the versions of the tools used to build the plugin.
+As of Jenkins 1.646, they've been decoupled:
+Plugin POM 2.0 and newer allow specifying the core version dependency independent of its version, so that even plugins compatible with older Jenkins releases can benefit from fixes and improvements in the parent POM.
+
+== Keeping up to date
+
+It is generally recommended to continually update to recent plugin parent POMs for multiple reasons:
+
+* Plugin developers and users benefit from bug fixes and improvements to the parent POM and the tools it specifies
```

Comment:
Updated test harness versions is another benefit. Though arguably that's covered by "the tools it specifies".